pilot/pkg/serviceregistry/federation/controller.go

```diff
@@ -383,10 +432,13 @@ func (c *Controller) handleEvent(e *federation.WatchEvent) {
 }
 
 func (c *Controller) watch(eventCh chan *federation.WatchEvent, stopCh <-chan struct{}) error {
-	req, err := http.NewRequest("GET", c.discoveryEndpoint+"/watch", nil)
+	url := c.dicsoveryURL + "/watch"
+	req, err := http.NewRequest(http.MethodGet, url, nil)
 	if err != nil {
-		return err
+		logger.Errorf("Failed to create request: '%s': %s", url, err)
+		return nil
 	}
+	req.Header.Add("discovery-address", c.networkAddress)
```

Comment:
why return the discovery address?

Reply:
I did this to support routing, in case the gateway is being used to route traffic to multiple external meshes.  The source of the traffic is istiod, which is not running a proxy and is just sending a raw (non-istio) http request.  Internally, the request is sent directly to the egress gateway with this header.  The header is used in a VirtualService to route to a specific discovery server in another mesh.